include/container_iterator.h

```diff
@@ -47,10 +57,20 @@ namespace reinforcement_learning {
     int64_t operator-(const container_iterator& rhs) const {
       return _idx - rhs._idx;
     }
+    //! Allow distance measurement
+    container_iterator operator-(const int idx) const {
```

Comment:
Do we need this?
Also, it is outside of scope of this PR, but I believe there is a size_t overflowing issue on lines 57-59 which we need to fix.

Reply:
Yes. There are required to sort the ranking array (action, probability). Check live_modeL_impl.cc post_process_rank.

Here I only added the minimum number of operator I need

Reply:
oh I see.

Reply:
Fixed line 57-59 to use size_t

Reply:
counterexample:
_idx = 1
rhs._idx = 2

_idx - rhs._idx will be first evaluated as size_t (to max(size_t) - 1) and then converted to return type.
So return type insignature should be left as int64_t, but return expression changed to:
static_cast<int64_t>(_idx) - static_cast<int64_t>(rhs._idx)

Reply:
Fixed.